src/coreclr/scripts/genEventPipe.py

```diff
@@ -283,9 +301,10 @@ def generateWriteEventBody(template, providerName, eventName, runtimeFlavor):
             if template.name in specialCaseSizes and paramName in specialCaseSizes[template.name]:
                 size = "(int)(%s)" % specialCaseSizes[template.name][paramName]
             if runtimeFlavor.mono:
-                pack_list.append(
-                    "    success &= write_buffer((const uint8_t *)%s, %s, &buffer, &offset, &size, &fixedBuffer);" %
-                    (paramName, size))
+                t = winTypeToFixedWidthType(parameter.winType)
```

Comment:
Thank you very much for the detailed review :)

My intention of having only one endian-independent implementation was that this will be better tested than having two obviously. Having a look at the assembly of a release build I see that for `EventPipeWriteEventMethodDCEndILToNativeMap` as well as `EventPipeWriteEventMonoProfilerJitDone_V1` all calls to `write_buffer_x` and `memcpy` are inlined (on x64 and s390x), however, this does not include the call to `write_buffer` from `eventpipehelpers.c` of course. Thus, in order to keep the overhead minimal and not introduce any potential performance regression I restored the old behavior for little-endian architectures.

Note, while having a further look at all `write_buffer_x` functions I changed `write_buffer_uintptr_t` slightly in case of `sizeof (uintptr_t) == 4`.